### ISSUE TITLE
Rosetta Api support for Network Status

### ISSUE THREAD

**Issue by noumanahmad001** [blockstack/stacks-blockchain#1799]
**Is your feature request related to a problem? Please describe.**
The Rosetta [/network/status](https://www.rosetta-api.org/docs/NetworkApi.html) call returns the networks's status along with a list of peers. Some of the information is available from the [sidecar's database](https://github.com/blockstack/stacks-blockchain-api/issues/187), but there is no RPC call implemented by the stacks-node that returns a list of peers.
The Rosetta Api response requires this Peer [object](https://www.rosetta-api.org/docs/models/Peer.html). I haven't found a peer id in the stacks-blockchain. 

**Describe the solution you'd like**
RPC endpoint that returns peer information. We can implement another RPC endpoint but we are having trouble finding the peer id.

**Comment by diwakergupta** [comment]
Does `/v2/neighbors` work for you? Note that the full set of peers is NOT enumerable by design in the Stacks network, and AFAICT the Rosetta spec doesn't set any expectations on whether this list is exhaustive or not. All that to say, the info in `/v2/neighbors` will have to do for now.

**Comment by fariedt** [comment]
We've had a deeper look into the Rosetta specs, and what it wants for the "peerid" is not the peers that the stacks node is connected with (the data that is returned by `/v2/neighbors`), but the node or nodes that the sidecar/rosetta server is connected to.  It is the public key of the node that the sidecar server talks with.  This information isn't present in `/v2/info` but I can generate a pull request for that change this week.

**Comment by diwakergupta** [comment]
> We've had a deeper look into the Rosetta specs, and what it wants for the "peerid" is not the peers that the stacks node is connected with (the data that is returned by `/v2/neighbors`), but the node or nodes that the sidecar/rosetta server is connected to. It is the public key of the node that the sidecar server talks with. This information isn't present in `/v2/info` but I can generate a pull request for that change this week.

@fariedt `v2/neighbors` already includes the `public_key_hash` of all authenticated peers. I'm not sure I follow the distinction you're making between "peers that the stacks node is connected to" and "nodes that the sidecar/rosetta server is connected to".

Again, by design, the set of all peers on the Stacks blockchain is NOT enumerable and anyways membership is dynamic as nodes come and leave the network. So at best, a single Stacks node (whether it's connected to the sidecar or not) can provide a point-in-time snapshot of the peers it knows about. This information is already included in the neighbors endpoint. Unless I'm missing something, I don't think any changes are needed to `v2/info`.

**Comment by lgalabru** [comment]
@fariedt per my understanding of https://www.rosetta-api.org/docs/NetworkApi.html, the json node `peers` is supposed to return the (volatile) set of peers (`/v2/neighbors`) that the stacks-node, inspected by your rosetta-server, is talking with (vs the stacks-node itself, as I think you're suggesting). You probably have a 1:1 relationship between a rosetta server and a stacks-node, so `peers` (plural) would be weird. Do you have additional documentation, demonstrating your suspicions?

**Comment by fariedt** [comment]
I looked over the Rosetta community forums for examples of implementations for other blockchains, and it does look like `/v2/neighbors` will return what they want.  I will withdraw my pull request in a bit.

A couple of implementations return `null` for that field, and I'm not sure if that's because they haven't implemented it yet -- [ckb](https://github.com/nervosnetwork/ckb-rosetta-sdk/) and [iotex](https://github.com/iotexproject/iotex-core-rosetta-gateway).  I don't remember which one it was, but another implementation's example had two peers with IP addresses of 127.0.0.1 and 127.0.0.2 -- probably output made while testing their implementation.  The Rosetta docs do not explain what the output of that call will be used for, and especially whether the nodes returned by that call should be reachable over the internet.

**Comment by diwakergupta** [comment]
@fariedt looks like we're not blocked on this. I'm going to close out, please reopen or open a new issue as appropriate, if needed.
